Se/swerve commands

Keobkeig opened the pull request:
SwerveDrivetoShoot doesnt work as intended
SwerveDriveXMode should be tested on real robot

Keobkeig committed:
Add SwerveDriveToPose.java and prep test it in sim

Co-authored-by: Tmanxyz <Tmanxyz@users.noreply.github.com>
Co-authored-by: metereel <Material-Energy@users.noreply.github.com>
Co-authored-by: alex <alex-wanger@users.noreply.github.com>
Co-authored-by: rzheng287 <rzheng287@users.noreply.github.com>

Keobkeig committed:
Add SwerveCommands

Co-authored-by: Tmanxyz <Tmanxyz@users.noreply.github.com>
Co-authored-by: metereel <Material-Energy@users.noreply.github.com>
Co-authored-by: alex <alex-wanger@users.noreply.github.com>
Co-authored-by: rzheng287 <rzheng287@users.noreply.github.com>

Keobkeig committed:
fix SwerveDriveDrive turn not working, but now wheels always pointing inward/parallel

Keobkeig committed:
add commands

Co-authored-by: naveed1117 <naveed1117@users.noreply.github.com>
Co-authored-by: jopy-man <jopy-man@users.noreply.github.com>
Co-authored-by: Tmanxyz <Tmanxyz@users.noreply.github.com>

anivanchen commented on src/main/java/com/stuypulse/robot/commands/swerve/SwerveDriveToShoot.java:
```diff
@@ -0,0 +1,113 @@
+    private Pose2d getSpeakerTargetPose(Rotation2d angleToSpeaker) {
+        //Everything under is from aiming at center of speaker
+        Vector2D robotPose = new Vector2D(Odometry.getInstance().getPose().getTranslation());
+        Vector2D targetPose = new Vector2D(this.targetPose.getTranslation());
```
Interpolate between target poses +- certain y distance from the center of the speaker to adjust the point which the robot will aim at for shooting.

anivanchen commented on src/main/java/com/stuypulse/robot/commands/swerve/SwerveDriveToShoot.java:
```diff
@@ -0,0 +1,113 @@
+    private final HolonomicController controller;
+    private final FieldObject2d targetPose2d;
+
+    public SwerveDriveToShoot(Pose2d targetPose) {
```
Remove unused targetPose param

anivanchen replied:
This command is supposed to align from any point on the field to a certain shooting distance away from the speaker and to point at the speaker target position. then shoot.

Keobkeig committed:
SwerveDriveToShoot works

Keobkeig committed:
Fix XMode and DriveToShoot

Keobkeig committed:
Fixed merge conflicts for pr

Merge branch 'main' of https://github.com/StuyPulse/BigWang into se/swerve-commands

BenG49 committed:
Remove testing button bindings

BenG49 committed:
Remove gyro feedback

BenG49 committed:
Merge branch 'main' of github.com:StuyPulse/BigWang into se/swerve-commands

BenG49 committed:
Comment Field.java

BenG49 committed:
Make x mode states constant

BenG49 committed:
Update swerve logging

BenG49 committed:
Remove Gyro constant from alignment settings

BenG49 committed:
Slight changes
